# 1)Marketing team from Xyz company wants to launch a product to the market. Their initial results shows the average salary of the people in x place is 50,000 taka which is their target customer. To prove their hypothesis they collected 20 different samples. Now let's run a hypothesis test to prove their statement.

So we need to.

1. Clearly define null and alternate hypothesis.
2. Run a statistical test.
3. Show the p value.
4. Prove the statement.

Average salary from 20 samples:

sample= [40000,45000,35000,60000,34000,15000,56000,44000,67000,34000,45000,34000,45000,90000,45000,45000,37000,45000,42000,55000,50000]

 confidense interval is 95%

# Null Hypothesis (H0): The average salary of the people in x place is not 50,000
# Alternate Hypothesis (HA): The average salary of the people in x place is 50,000

In [1]:
#importing necessary libraries
import scipy.stats as stats
import math 
%matplotlib inline

In [2]:
population= [30000, 45000, 55000,65000,70000,25000,20000,37000,67000,38000,67000,55000,54000,18000,57000,120000,12000,22000,65000, 32000,34000]
sample= [40000,45000,35000,60000,34000,15000,56000,44000,67000,34000,45000,34000,45000,90000,45000,45000,37000,45000,42000,55000,50000]
from statistics import mean
print( 'Average sallary of population ', mean(population))
print('Average sallary of Sample ', mean(sample))
stats.ttest_1samp(a=sample, popmean=mean(population))

Average sallary of population  47047.619047619046
Average sallary of Sample  45857.142857142855


Ttest_1sampResult(statistic=-0.3667006002656119, pvalue=0.7176933249465037)

hence confidence interval is 95% so, p value should be less then or equal 0.05 but reslted P-value is 0.7177 means p value is greater than 0.05 ,So we can not reject null Hypothesis. So statement provided by marketing team is not proven

# 2) Suppose we want to test if a particular drug is effective to reduce weight. Now we have to test our hypothesis to prove our statement.

sample before drug test= [302.48364224, 241.4178101 , 235.46304603, 170.40044322,
       249.75146112, 240.41105909, 233.90111913, 259.46208005,
       262.63152149, 218.03191059, 223.4128099 , 235.72799522,
       270.69046931, 266.83576528, 210.83354471, 216.41574215,
       272.10512174, 297.23902219, 249.06774735, 229.49660118,
       282.86889096, 240.71270088, 271.77256674, 296.472149  ,
       268.90239467, 252.20479711, 271.96814062, 230.72273823,
       244.65720474, 232.78136322]

sample after drug test= [300.21176566, 237.73533459, 233.28416941, 167.24776114,
       248.9463493 , 239.47941739, 234.13285467, 265.22593565,
       253.64720865, 223.26000351, 220.97655767, 228.31626419,
       268.57837047, 266.04495716, 214.92133699, 209.85756991,
       271.94186278, 296.57811971, 239.39719288, 222.31782485,
       284.61940098, 242.94053716, 275.96112104, 297.89125758,
       269.62845474, 251.56913476, 276.76364883, 225.25740772,
       242.69773684, 233.45813391]

we need to.

1. Clearly define null and alternate hypothesis.
2. Run a statistical test.
3. Show the p value.
4. Prove our statement.

consider confidense interval is 90%


# Null Hypothesis (H0): There is no relationship between drug and Weight
# Alternate Hypothesis (HA): Drug is effective to Reduce weight

In [3]:
alpha= 0.1
sample_before_drug_test= [302.48364224, 241.4178101 , 235.46304603, 170.40044322, 249.75146112, 240.41105909, 233.90111913, 259.46208005, 262.63152149, 218.03191059, 223.4128099 , 235.72799522, 270.69046931, 266.83576528, 210.83354471, 216.41574215, 272.10512174, 297.23902219, 249.06774735, 229.49660118, 282.86889096, 240.71270088, 271.77256674, 296.472149 , 268.90239467, 252.20479711, 271.96814062, 230.72273823, 244.65720474, 232.78136322]
sample_after_drug_test=[300.21176566, 237.73533459, 233.28416941, 167.24776114, 248.9463493 , 239.47941739, 234.13285467, 265.22593565, 253.64720865, 223.26000351, 220.97655767, 228.31626419, 268.57837047, 266.04495716, 214.92133699, 209.85756991, 271.94186278, 296.57811971, 239.39719288, 222.31782485, 284.61940098, 242.94053716, 275.96112104, 297.89125758, 269.62845474, 251.56913476, 276.76364883, 225.25740772, 242.69773684, 233.45813391]

In [4]:
t_value, p_value=stats.ttest_rel(sample_before_drug_test,sample_after_drug_test )
one_tailed_p_value=float("{:.6f}".format(p_value/2))
print('Test statistics is %f'%float("{:.6f}".format(t_value)))
print('p-value for one_tailed_test is %f'%one_tailed_p_value)
alpha=0.1
if one_tailed_p_value<=alpha:
    print('Conclusion:','Hence p_value(=%f)'%one_tailed_p_value,'<' 'alpha(=%.2f)'%alpha,'''We reject the null hypothesis H0
    So, wc can conclude that drug is effective to reduce weight''')
else:
    print('Conclusion:','Hence p_value(=%f)'%one_tailed_p_value,'>' 'alpha(=%.2f)'%alpha,'''We can not reject the null hypothesis H0
    So, wc can conclude that drug is not effective to reduce weight ,that means there is no relationship between drug and weight''')

Test statistics is 1.596886
p-value for one_tailed_test is 0.060566
Conclusion: Hence p_value(=0.060566) <alpha(=0.10) We reject the null hypothesis H0
    So, wc can conclude that drug is effective to reduce weight
